code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,918 @@
+	..()
+	if(!istype(AM,/mob/living))
+		return
+	if(locate(holotile) in T)//here to both delete the previous hologram,
+		qdel(holotile)
+	L = AM
+	holotile = new(T) //setting up the hologram to look like the person that just stepped in
+	holotile.icon = L.icon
+	holotile.icon_state = L.icon_state
+	holotile.alpha = 100
+	holotile.name = "[L.name] (Hologram)"
+	holotile.add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
+	holotile.copy_overlays(L, TRUE)
+
+///destroys the hologram with the rune
+/obj/effect/warped_rune/ceruleanspace/Destroy()
+	..() //needed so the rune still gets stored by the crossbreed
+	qdel(holotile)
+
+
+
+
+
+/obj/item/slimecross/warping/pyrite
+	colour = "pyrite"
+	runepath = /obj/effect/warped_rune/pyritespace
+	effect_desc = "draws a rune that will randomly color whoever steps on it"
+
+/obj/effect/warped_rune/pyritespace
+	desc = "Who shall we be today? they asked, but not even the canvas would answer."
+	icon_state = "colorune"
+
+
+///colors whoever steps on the rune randomly
+/obj/effect/warped_rune/pyritespace/Crossed(atom/movable/AM)
+	..()
+	AM.color = rgb(rand(0,255),rand(0,255),rand(0,255))
+
+
+/* Will make anyone on the rune do much more punch damage. Doesn't boost normal melee weapon damage, may need rebalancing,*/
+
+/obj/item/slimecross/warping/red
+	colour = "red"
+	runepath = /obj/effect/warped_rune/redspace
+	effect_desc = "Draws a rune severely increasing your punch damage as long as you stand on it."
+	drawing_time = 100 // slightly longer to draw so you don't draw it in the middle of a fight.
+
+/obj/effect/warped_rune/redspace
+	desc = "Progress is made through adversity, power is obtained through violence"
+	icon_state = "rage_rune"
+	var/mob/living/carbon/human/H
+
+
+///boost up the unarmed damage of the person currently on the tile.
+/obj/effect/warped_rune/redspace/Crossed(atom/movable/AM)
+	..()
+	if(!istype(AM,/mob/living/carbon/human))
+		return
+	H = AM
+	H.dna.species.punchstunthreshold += 20//we don't want them to insta stun everyone.
+	H.dna.species.punchdamagelow += 20
+	H.dna.species.punchdamagehigh += 20 //buffed up punch damage
+	to_chat(H, "<span class='warning'>You feel the urge to punch something really hard!</span>")
+
+
+///takes away the punch damage when you leave
+/obj/effect/warped_rune/redspace/Uncrossed(atom/movable/AM)
+	if(!istype(AM,/mob/living/carbon/human)) //checks if the person that just left is the same as the currently enraged person
+		return
+	H = AM
+	H.dna.species.punchstunthreshold -= 20
+	H.dna.species.punchdamagelow -= 20
+	H.dna.species.punchdamagehigh -= 20
+
+
+///destroying the rune will also remove the punch force of the persons on the rune.
+/obj/effect/warped_rune/redspace/Destroy()
+	..()
+	for(var/mob/living/carbon/human/D in T) // takes away people's punch damage that are on the rune anyway.
+		H.dna.species.punchstunthreshold -= 20
+		H.dna.species.punchdamagelow -= 20
+		H.dna.species.punchdamagehigh -= 20.
```

Comment:
Always `return ..()` at the end of `Destroy()` procs.
Preserve the return and do it at the end of the proc.